hefquin-cli/src/main/java/se/liu/ida/hefquin/cli/modules/ModQuery.java

```diff
@@ -1,21 +1,60 @@
 package se.liu.ida.hefquin.cli.modules;
 
+import org.apache.jena.cmd.ArgDecl;
+import org.apache.jena.cmd.CmdArgModule;
 import org.apache.jena.cmd.CmdGeneral;
+import org.apache.jena.cmd.ModBase;
+import org.apache.jena.query.Query;
+import org.apache.jena.query.QueryFactory;
 import org.apache.jena.query.Syntax;
+import org.apache.jena.shared.JenaException;
+import org.apache.jena.shared.NotFoundException;
 
-import arq.cmdline.ModQueryIn;
-
-public class ModQuery extends ModQueryIn
+/**
+ * Command-line argument module for specifying query-related arguments.
+ */
+public class ModQuery extends ModBase
```

Comment:
What is the reason for changing the superclass of this class from `ModQueryIn` to `ModBase`?

Reply:
`ModQueryIn` assumes some functionality that we (currently) do not support:
- specifying the query syntax (`--syntax`)
- passing the query as a string
- reading the query from `System.in`

This also makes the output a bit unrepresentative (e.g., "No query string or query file" instead of "No query file")

Reply:
> `ModQueryIn` assumes some functionality that we (currently) do not support: [...]

Okay. Thanks for the clarification!